diff -ru src/knemod/backends/sysbackend.cpp knemod/backends/sysbackend.cpp
--- src/knemod/backends/sysbackend.cpp      2007-05-24 20:52:02.000000000 +0200
+++ src/knemod/backends/sysbackend.cpp      2013-06-13 23:43:48.017512336 +0200
@@ -327,7 +327,8 @@
             struct iw_range range;
             if( iw_get_range_info( fd, ifName.latin1(), &range ) >= 0 )
             {
-                if ( freq < KILO )
+                //if ( freq < KILO )
+                if ( freq < 1e3 )
                 {
                     channel = iw_channel_to_freq( (int) freq, &freq, &range );
                 }
