--- avahi-utils/Makefile.am
+++ avahi-utils/Makefile.am
@@ -52,6 +52,7 @@
 avahi_set_host_name_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la
 
 install-exec-local:
+	[[ ! -d $(DESTDIR)/$(bindir) ]] && mkdir -p $(DESTDIR)/$(bindir)
 	cd $(DESTDIR)/$(bindir) && \
 		rm -f avahi-resolve-host-name avahi-resolve-address avahi-browse-domains avahi-publish-address avahi-publish-service && \
 		$(LN_S) avahi-resolve avahi-resolve-host-name && \
