--- kernel/nvidia/nv-procfs.c	2015-10-11 23:09:35.062904349 +0300
+++ kernel/nvidia/nv-procfs.c	2015-10-11 23:10:01.904903762 +0300
@@ -360,7 +360,8 @@
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
-    return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
 }
 
 static ssize_t
@@ -560,7 +561,8 @@
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
