--- intern/itasc/SConscript	2009-09-26 09:14:02.000000000 +0200
+++ intern/itasc/SConscript	2009-09-26 09:14:54.000000000 +0200
@@ -5,7 +5,8 @@
 sources += env.Glob('kdl/*.cpp')
 sources += env.Glob('kdl/utilities/*.cpp')
 
-incs = '. ../../extern/Eigen3'
+#incs = '. ../../extern/Eigen3'
+incs = '. /usr/include/eigen3'
 
 env.BlenderLib ('bf_intern_itasc', sources, Split(incs), [], libtype=['intern','player'], priority=[20,100] )  
 
--- source/blender/ikplugin/SConscript
+++ source/blender/ikplugin/SConscript
@@ -4,6 +4,6 @@
 sources = env.Glob('intern/*.c') + env.Glob('intern/*.cpp')
 
 incs = '#/intern/guardedalloc #/intern/iksolver/extern ../makesdna ../blenlib'
-incs += ' ../blenkernel ../include ../ikplugin #/intern/itasc #/extern/Eigen3'
+incs += ' ../blenkernel ../include ../ikplugin #/intern/itasc /usr/include/eigen3'
 
 defs.append('WITH_IK_ITASC')
--- source/gameengine/Converter/SConscript	2011-08-30 19:56:07.755000018 -0600
+++ source/gameengine/Converter/SConscript	2011-08-30 19:54:54.238000019 -0600
@@ -20,7 +20,7 @@
 incs += ' #source/blender/makesrna'
 incs += ' #source/blender/ikplugin'
 incs += ' #extern/recastnavigation/Detour/Include'
-incs += ' #extern/Eigen3'
+incs += ' /usr/include/eigen3'
 
 incs += ' ' + env['BF_BULLET_INC']
 
