--- /home/rainman/src/kde/kate-plugins-3.5.10/kate/tabbarextension/plugin_katetabbarextension_old.h
+++ /home/rainman/src/kde/kate-plugins-3.5.10/kate/tabbarextension/plugin_katetabbarextension.h
@@ -53,7 +53,8 @@
 class MyPtrList : public QPtrList <KateTabBarButton>
 {
   public:
-  MyPtrList() { QPtrList<KateTabBarButton>::QPtrList(); }
+  //MyPtrList() { QPtrList<KateTabBarButton>::QPtrList(); }
+  MyPtrList() :QPtrList() { /* - */ }
 
   virtual int compareItems ( QPtrCollection::Item item1_, QPtrCollection::Item item2_ );
 };