diff --git a/HLLib/Makefile b/HLLib/Makefile
index 3fe2a76..ff8a019 100644
--- a/HLLib/Makefile
+++ b/HLLib/Makefile
@@ -21,11 +21,11 @@ clean:
 	rm -f \#* *~ *.o *.so.*
 
 install: libhl.so.$(HLLIB_VERS)
-	install -g root -m 0755 -o root -d $(PREFIX)/lib $(PREFIX)/include
-	install -g root -m 0644 -o root libhl.so.$(HLLIB_VERS) $(PREFIX)/lib
-	install -g root -m 0644 -o root ../lib/HLLib.h $(PREFIX)/include/hl.h
-	ln -fs $(PREFIX)/lib/libhl.so.$(HLLIB_VERS) $(PREFIX)/lib/libhl.so.2
-	ln -fs $(PREFIX)/lib/libhl.so.$(HLLIB_VERS) $(PREFIX)/lib/libhl.so
+	install -g root -m 0755 -o root -d ${DESTDIR}/$(PREFIX)/lib ${DESTDIR}/$(PREFIX)/include
+	install -g root -m 0644 -o root libhl.so.$(HLLIB_VERS) ${DESTDIR}/$(PREFIX)/lib
+	install -g root -m 0644 -o root ../lib/HLLib.h ${DESTDIR}/$(PREFIX)/include/hl.h
+	ln -fs ${DESTDIR}/$(PREFIX)/lib/libhl.so.$(HLLIB_VERS) ${DESTDIR}/$(PREFIX)/lib/libhl.so.2
+	ln -fs ${DESTDIR}/$(PREFIX)/lib/libhl.so.$(HLLIB_VERS) ${DESTDIR}/$(PREFIX)/lib/libhl.so
 
 libhl.so.$(HLLIB_VERS): $(objs)
 	$(CXX) $(LDFLAGS) -o $@ $(objs)
